packages/core/src/cells/loading-cell.tsx

```diff
@@ -1,13 +1,44 @@
+import { withAlpha } from "../internal/data-grid/color-parser.js";
+import { roundedRect } from "../internal/data-grid/data-grid-lib.js";
 import { GridCellKind, type LoadingCell } from "../internal/data-grid/data-grid-types.js";
 import type { InternalCellRenderer } from "./cell-types.js";
 
+function getRandomNumber(x: number, y: number): number {
+    let seed = x * 49_632 + y * 325_176;
+
+    // Inline Xorshift algorithm
+    seed ^= seed << 13;
+    seed ^= seed >> 17;
+    seed ^= seed << 5;
+
+    // eslint-disable-next-line unicorn/number-literal-case
+    return (seed / 0xff_ff_ff_ff) * 2;
+}
+
 export const loadingCellRenderer: InternalCellRenderer<LoadingCell> = {
     getAccessibilityString: () => "",
     kind: GridCellKind.Loading,
     needsHover: false,
     useLabel: false,
     needsHoverPosition: false,
     measure: () => 120,
-    draw: () => undefined,
+    draw: a => {
+        const { cell, col, row, ctx, rect, theme } = a;
+        if (cell.skeletonWidth === undefined || cell.skeletonWidth === 0) {
+            return;
+        }
+
+        let width = cell.skeletonWidth;
+        if (cell.skeletonVariability !== undefined && cell.skeletonVariability > 0) {
+            width += Math.round(getRandomNumber(col, row) * cell.skeletonVariability);
+        }
+
+        const hpad = theme.cellHorizontalPadding;
+        const rectHeight = 18;
+
+        roundedRect(ctx, rect.x + hpad, rect.y + (rect.height - rectHeight) / 2, width, rectHeight, 3);
```

Comment:
Not really something to change in this PR, but I think the "roundness" could also be considered a theming aspect. Maybe it makes sense to introduce a single rounding radius theming option which is used as a default/fallback for rounded aspects like checkbox, image, drilldown, button, and tags cell.

Reply:
This is a good idea in general.